### ISSUE TITLE
Opacity issues on Kubuntu

### ISSUE THREAD

**Issue by catalyst1987** [alacritty/alacritty#3759]
Had issues with setting opacity.  No matter what I set it had no effect.  I then changed my compositor from OpenGL to XRender.  Then Alacritty became completely transparent and when I try to change the opacity it seems to have no effect.  

### System

OS: Kubuntu 20.04
Version: 0.4.2
Linux/BSD: X11

**Comment by catalyst1987** [comment]
Here is the contents of alacritty -vvv


[AlacrittyDebug.txt](https://github.com/alacritty/alacritty/files/4670002/AlacrittyDebug.txt)

**Comment by catalyst1987** [comment]
Here is a screenshot and also my config file.  
[alacritty.txt](https://github.com/alacritty/alacritty/files/4670012/alacritty.txt)



![image](https://user-images.githubusercontent.com/26027600/82703047-17072780-9c6b-11ea-8591-bf6ce18a0427.png)

**Comment by chrisduerr** [comment]
Can you share your Alacritty configuration?

**Comment by kchibisov** [comment]
>Can you share your Alacritty configuration?

>Here is a screenshot and also my config file alacritty.txt

It was shared already.

**Comment by kchibisov** [comment]
Remove leading white space before `background_opacity`.

**Comment by catalyst1987** [comment]
> Remove leading white space before `background_opacity`.

If I remove the space, I get an error on the configuration.  See attached.  
[alacritty.txt](https://github.com/alacritty/alacritty/files/4670496/alacritty.txt)
[Alacritty-4440.log](https://github.com/alacritty/alacritty/files/4670499/Alacritty-4440.log)

**Comment by chrisduerr** [comment]
You need to delete the comment character `#` before fields you want to use, not replace it with a space. Your config does not have correct indentation.

**Comment by catalyst1987** [comment]
> Can you share your Alacritty configuration? Also how did you change your compositor, is that a KDE setting?

The compositor was originally set to OpenGL 2.0.  At that point there was no transparency at all no matter what the opacity was set to.  I then changed the compositor to XRender and now alacritty seems to be fully transparent, again regardless of what I set the opacity too.

**Comment by kchibisov** [comment]
Try this updated config. YAML file is indention sensitive, so you should just delete `#` in our config(we try to keep it properly indented that way), not replace with spaces. 

[alacritty.yml.txt](https://github.com/alacritty/alacritty/files/4670513/alacritty.yml.txt)

**Comment by catalyst1987** [comment]
> Try this updated config. YAML file is indention sensitive, so you should just delete `#` in our config(we try to keep it properly indented that way), not replace with spaces.
> 
> [alacritty.yml.txt](https://github.com/alacritty/alacritty/files/4670513/alacritty.yml.txt)

So I downloaded it and ran:

`cp  /home/catalyst/Downloads/alacritty.yml.txt /home/catalyst/.config/alacritty/alacritty.yml`

No errors when running the terminal, but still the same issue.  

![image](https://user-images.githubusercontent.com/26027600/82711956-be418a00-9c7e-11ea-9d8f-a264727d63c2.png)

I am beginning to wonder if my KDE settings are overriding the terminal opacity settings.

**Comment by kchibisov** [comment]
I guess your compositor is failing, since it works properly now for me on a system with native transparency(Wayland). Try to change settings of it or opacity setting to higher values(not higher than 1, since it's capped anyway).

**Comment by catalyst1987** [comment]
> I guess your compositor is failing, since it works properly now for me on a system with native transparency(Wayland). Try to change settings of it or opacity setting to higher values(not higher than 1, since it's capped anyway).

No change unfortunately. 

Following my hunch that it has something to do with KDE plasma I ran the following command

`kquitapp5 plasmashell`

Upon which my terminal went completely opaque.  If I restart KDE plasma using:

`kstart5 plasmashell`

The terminal goes completely transparent again.

**Comment by catalyst1987** [comment]
> > I guess your compositor is failing, since it works properly now for me on a system with native transparency(Wayland). Try to change settings of it or opacity setting to higher values(not higher than 1, since it's capped anyway).
> 
> No change unfortunately.
> 
> Following my hunch that it has something to do with KDE plasma I ran the following command
> 
> `kquitapp5 plasmashell`
> 
> Upon which my terminal went completely opaque. If I restart KDE plasma using:
> 
> `kstart5 plasmashell`
> 
> The terminal goes completely transparent again.

Disregard this.  I thought the terminal went completely opaque but the desktop background simply went black when I killed plamsa.  So I guess thats not the issue either.

**Comment by kchibisov** [comment]
Is it always transparent regardless of settings?

**Comment by catalyst1987** [comment]
> Is it always transparent regardless of settings?

Seems to be the case, yes.

**Comment by kchibisov** [comment]
Can you try `alacritty --config-file /dev/null` is it also transparent?

**Comment by catalyst1987** [comment]
> Can you try `alacritty --config-file /dev/null` is it also transparent?

Yes, it is.

Another interesting thing to note is that if I right click on the window title bar for alacritty and then go to "More Actions" > "Configure Special Application Settings" > "Appearance & Fixes" and then tick the option to "Block compositing" and set the option to "Force" and tick the radio button for "yes", the terminal will then be completely opaque, but then making any changes to the yml file for opacity have no effect.

**Comment by kchibisov** [comment]
Also, could you check that you don't have any settings related to opacity for alacritty outside of the config, IIRC some compositors allow you to set opacity per window, the thing is that your text is also transparent, so it seems like opacity wasn't done by us or someone else is trying to do it in addition.

**Comment by catalyst1987** [comment]
> Also, could you check that you don't have any settings related to opacity for alacritty outside of the config, IIRC some compositors allow you to set opacity per window, the thing is that your text is also transparent, so it seems like opacity wasn't done by us or someone else is trying to do it in addition.

I managed to fix this.  I did the following in case it helps anyone who encounters similar issue and might find this helpful.

1)  Went into the NVIDIA X Server Settings and the Antialiasing Settings and unticked the "Enable FXAA" option.
2)  Went into the Compositor settings and tried going back to OpenGL 3.1 and also set the scale method to Smooth.  

Not only by doing this did some strange screen flickering that was happening when the compositor was set to OpenGL finally went away, but the opacity settings in the YML file now work.  :-)

Many Thanks for all the assistance on this one.  I really appreciate it.
